src/Parser/Ast/ExpressionNode.php

```diff
@@ -122,8 +123,14 @@ public static function fromTokens(\Iterator $tokens, Precedence $precedence = Pr
         }
 
         Scanner::skipSpaceAndComments($tokens);
+        if (Scanner::isEnd($tokens) || $precedence->mustStopAt(Scanner::type($tokens))) {
+            return new self(
+                root: $root
+            );
+        }
 
         while (!Scanner::isEnd($tokens) && !$precedence->mustStopAt(Scanner::type($tokens))) {
+            Scanner::skipSpaceAndComments($tokens);
```

Comment:
The additions that are here are not necessary. The condition is also handled by the "while" loop and later comes the return. 

I think i messed stuff up because the partial fix belongs to https://github.com/PackageFactory/PackageFactory.ComponentEngine/pull/10

either way, it works as it is, just not super clean.